pkg/ruler/api_test.go

```diff
@@ -1630,32 +1630,33 @@ func TestAPIRoutesCorrectlyHandleInvalidOrgID(t *testing.T) {
 	}
 
 	for _, tc := range tcs {
-		const invalidOrgID = ""
-
-		cfg := defaultRulerConfig(t)
-		cfg.TenantFederation.Enabled = true
-
-		r := prepareRuler(t, cfg, newMockRuleStore(map[string]rulespb.RuleGroupList{}), withStart())
-
-		a := NewAPI(r, r.store, log.NewNopLogger())
-
-		router := mux.NewRouter()
-		router.Path("/api/v1/rules").Methods(http.MethodGet).HandlerFunc(a.PrometheusRules)
-		router.Path("/api/v1/alerts").Methods(http.MethodGet).HandlerFunc(a.PrometheusAlerts)
-		router.Path("/config/v1/rules").Methods(http.MethodGet).HandlerFunc(a.ListRules)
-		router.Path("/config/v1/rules/{namespace}").Methods(http.MethodGet).HandlerFunc(a.ListRules)
-		router.Path("/config/v1/rules/{namespace}/{groupName}").Methods(http.MethodGet).HandlerFunc(a.GetRuleGroup)
-		router.Path("/config/v1/rules/{namespace}").Methods(http.MethodPost).HandlerFunc(a.CreateRuleGroup)
-		router.Path("/config/v1/rules/{namespace}/{groupName}").Methods(http.MethodDelete).HandlerFunc(a.DeleteRuleGroup)
-		router.Path("/config/v1/rules/{namespace}").Methods(http.MethodDelete).HandlerFunc(a.DeleteNamespace)
-
-		req := requestFor(t, tc.method, "https://localhost:8080"+tc.route, nil, invalidOrgID)
-
-		w := httptest.NewRecorder()
-		router.ServeHTTP(w, req)
+		for _, tenantID := range []string{"", "user1|user2"} {
+			t.Run(fmt.Sprintf("method=%s, route=%s, userID=%s", tc.method, tc.route, tenantID), func(t *testing.T) {
```

Comment:
> `userID=%s", ... tenantID`

(nit)

1. the test is called `...HandleInvalidOrgID`
2. we define tenant IDs
3. we name a sub-test `userID=blah`

I know that it's a historical mess that we use `OrgID`, `TenantID` and `UserID` for the same thing. But, I wonder if we could keep those names scoped within one test function.